backend/handler/answers.go

```diff
@@ -328,3 +328,48 @@ func (s *Server) PutUserAnswer(e echo.Context, traqID TraqId, questionID Questio
 
 	return e.JSON(http.StatusOK, res)
 }
+
+func (s *Server) GetQuestionAnswers(e echo.Context, questionID QuestionId, params GetQuestionAnswersParams) error {
+	user, err := s.repo.GetOrCreateUser(*params.XForwardedUser)
+	if err != nil {
+		e.Logger().Errorf("failed to get or create user: %v", err)
+
+		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+	}
+
+	targetQuestion, err := s.repo.GetQuestionByID(uint(questionID))
+	if err != nil {
+		e.Logger().Errorf("failed to get question: %v", err)
+
+		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+	}
+
+	if !targetQuestion.IsOpen && !user.IsStaff { // 公開じゃない場合はスタッフ以外は見れない
+		return echo.NewHTTPError(http.StatusForbidden, "Forbidden")
+	}
+
+	answers, err := s.repo.GetAnswersByID(uint(questionID))
+	if err != nil {
+		e.Logger().Errorf("failed to get answers: %v", err)
+		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+	}
+
+	var res GetQuestionAnswers
+
+	questionIDInt := int(questionID)
+	res.QuestionId = &questionIDInt
+	res.Answers = new([]interface{})
+
+	for _, answer := range answers {
+		answeredUesr, err := s.repo.GetUserTraqID(answer.UserID)
+		if err != nil {
+			e.Logger().Errorf("failed to get user: %v", err)
+
+			return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+		}
+
+		*res.Answers = append(*res.Answers, answeredUesr)
```

Comment:
スライスのサイズが分かっている場合は`make`で指定したほうが良いです
参考：https://zenn.dev/mkosakana/articles/2867cb1667bfa8

あと、各`answer`に対して`s.repo.GetUserTraqID`を呼ぶことでN+1問題が起きています。最初からtraQ IDを外部キーにしていればこんなことは起きなかったので原因は僕にあるし、修正にはDBのマイグレーションが必要になりそうなので今は放置で大丈夫なんですが、一応知っておいてほしいです :pray:
```suggestion
	res.Answers = make([]string, len(answers))

	for i, answer := range answers {
		answeredUser, err := s.repo.GetUserTraqID(answer.UserID)
		if err != nil {
			e.Logger().Errorf("failed to get user: %v", err)

			return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
		}

		res.Answers[i] = answeredUser
```

Reply:
ほんとですね、traqid毎回呼び出すこれは良くなさそう。初めて自分でn+1を作成してしまいました...